src/runtime/utils.ts

```diff
@@ -1,14 +1,47 @@
-import type { EventHandlerRequest, EventHandlerResponse, EventHandler } from "h3";
-import * as api from "@opentelemetry/api" 
-import { defineEventHandler } from "h3";
+import type {
+  EventHandlerRequest,
+  EventHandlerResponse,
+  EventHandler,
+  EventHandlerObject,
+} from "h3"
+import * as api from "@opentelemetry/api"
+import { defineEventHandler } from "h3"
 
 const context = api.context
 
 export function defineTracedEventHandler<
   Request extends EventHandlerRequest = EventHandlerRequest,
   Response = EventHandlerResponse,
->(handler: EventHandler<Request, Response>) {
-  return defineEventHandler<Request, Response>((event) => {
-    return context.with(event.otel.ctx, handler, undefined,  event) 
-  })
+>(handler: EventHandler<Request, Response> | EventHandlerObject<Request, Response>) {
+  if (isEventHandler(handler)) {
+    return defineEventHandler<Request, Response>((event) => {
+      return context.with(event.otel.ctx, handler, undefined, event)
+    })
+  } else if (isEventHandlerObject(handler)) {
+    const { handler: h, ...rest } = handler
+    return defineEventHandler({
+      ...rest,
+      handler: (event) => {
+        return context.with(event.otel.ctx, h, undefined, event)
+      },
+    })
+  }
```

Comment:
Would you like an `else`-case here that throws?

Reply:
else case ? 
I was thinking about adding `onRequest` and `onBeforeResponse` to the destructuring list so we can wrap them (only if it exists)

Reply:
I was talking about the `if-elseif` case that is there already. In case Nitro adds further patterns to the handler apart from `EventHandler` and `EventHandlerObject`. We only explicitly handle these two now, so should we have an `else` case that throws if an unexpected handler is passed?

Reply:
I'm still looking at the `onRequest`/`onBeforeResponse` handling separately. This was just a "spur of the moment" thought while looking at the current solution.

Reply:
oops my bad. I was refereing to 
> I don't think this will trace the onRequest and onBeforeResponse handlers

Reply:
I don't expect more pattern in Nitro V3 and I was told that OTEL is going to be implemented directly into nitro.

We can keep this as it is.

Reply:
Okay, let's keep it like it is then.

Seeing how smooth the new experimental OTEL support for Deno is, I really wanted to see the same for Nitro. Have you spoken with Pooya about OTEL or was it through some forum?

Reply:
Yes, I did speak with Pooya about that (hence the small notice about Nitro's implementation in the README.md :p )